compiler/src/dmd/expressionsem.d

```diff
@@ -14274,6 +14012,25 @@ Expression binSemanticProp(BinExp e, Scope* sc)
 // entrypoint for semantic ExpressionSemanticVisitor
 Expression expressionSemantic(Expression e, Scope* sc)
 {
+    if (e.type)
+    {
+        // When the expression has a type, it usually means that semantic
+        // has already been run yet.
+        // For these expressions, expressionsemantic is not idempotent yet,
+        // and the test suite fails without these. TODO: fix the code/tests
```

Comment:
Are you expecting to to this TODO in this PR?

Reply:
No, they probably should be done one at a time to make potential regressions easier to bisect.